tests/PHPStan/Parser/TypeParserTest.php

```diff
@@ -913,7 +913,7 @@ public function provideParseData(): array
 					],
 					new IdentifierTypeNode('C'),
 					[
-						new CallableTypeTemplateNode(new IdentifierTypeNode('A'), null),
+						new TemplateTagValueNode('A', null, ''),
```

Comment:
I would prefer to change https://github.com/phpstan/phpdoc-parser/blob/1.25.0/src/Ast/PhpDoc/TemplateTagValueNode.php#L15 first to keep `IdentifierTypeNode` instead of `string` to allow to store enriched attributes like position in the source phpdoc.

Reply:
changing `TemplateTagValueNode` would be a BC break though, and any other solution would require not reusing what already exists